libminifi/test/archive-tests/CompressContentTests.cpp

```diff
@@ -932,3 +938,107 @@ TEST_CASE("DecompressFileXYLZMA", "[compressfiletest8]") {
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::processors::PutFile>();
+
+  // Create temporary directories
+  char format_src[] = "/tmp/archives.XXXXXX";
+  std::string src_dir = testController.createTempDirectory(format_src);
+  REQUIRE(!src_dir.empty());
+
+  char format_dst[] = "/tmp/archived.XXXXXX";
+  std::string dst_dir = testController.createTempDirectory(format_dst);
+  REQUIRE(!dst_dir.empty());
+
+  // Define files
+  std::string src_file = utils::file::FileUtils::concat_path(src_dir, "src.txt");
+  std::string compressed_file = utils::file::FileUtils::concat_path(dst_dir, "src.txt.gz");
+  std::string decompressed_file = utils::file::FileUtils::concat_path(dst_dir, "src.txt");
+
+  // Build MiNiFi processing graph
+  auto plan = testController.createPlan();
+  auto get_file = plan->addProcessor(
+      "GetFile",
+      "GetFile");
+  auto compress_content = plan->addProcessor(
+      "CompressContent",
+      "CompressContent",
+      core::Relationship("success", "d"),
+      true);
+  auto put_compressed = plan->addProcessor(
+      "PutFile",
+      "PutFile",
+      core::Relationship("success", "d"),
+      true);
+  auto decompress_content = plan->addProcessor(
+      "CompressContent",
+      "CompressContent",
+      core::Relationship("success", "d"),
+      true);
+  auto put_decompressed = plan->addProcessor(
+      "PutFile",
+      "PutFile",
+      core::Relationship("success", "d"),
+      true);
+
+  // Configure GetFile processor
+  plan->setProperty(get_file, "Input Directory", src_dir);
+
+  // Configure CompressContent processor for compression
+  plan->setProperty(compress_content, "Mode", MODE_COMPRESS);
+  plan->setProperty(compress_content, "Compression Format", COMPRESSION_FORMAT_GZIP);
+  plan->setProperty(compress_content, "Update Filename", "true");
+  plan->setProperty(compress_content, "Encapsulate in TAR", "false");
+
+  // Configure first PutFile processor
+  plan->setProperty(put_compressed, "Directory", dst_dir);
+
+  // Configure CompressContent processor for decompression
+  plan->setProperty(decompress_content, "Mode", MODE_DECOMPRESS);
+  plan->setProperty(decompress_content, "Compression Format", COMPRESSION_FORMAT_GZIP);
+  plan->setProperty(decompress_content, "Update Filename", "true");
+  plan->setProperty(decompress_content, "Encapsulate in TAR", "false");
+
+  // Configure second PutFile processor
+  plan->setProperty(put_decompressed, "Directory", dst_dir);
+
+  // Create source file
+  std::string content;
+  SECTION("Empty content") {
+  }
+  SECTION("Short content") {
+    content = "Repeated repeated repeated repeated repeated stuff.";
+  }
+  SECTION("Long content") {
+    std::stringstream content_ss;
+    for (size_t i = 0U; i < 1024 * 1024U; i++) {
+      content_ss << "foobar";
+    }
+    content = content_ss.str();
+  }
+
+  std::fstream file;
+  file.open(src_file, std::ios::out);
+  file << content;
+  file.close();
```

Comment:
Could be a oneliner: `std::ofstream{src_file} << content;`. Just an idea, I don't mind if it stays like this.

Reply:
This is a test, I copied this from somewhere, I could not really be bothered whether its one line or four.